### PR TITLE
Kill webpack.

### PR DESCRIPTION
**Pull request by Mossop**
@victorporof this is what getting rid of webpack roughly looks like. It fails eslint tests but it runs and works. It moves a lot of stuff to make it work so I'm interested in what you think of the final directory structure.

### CODE REVIEW

**Comment by Mossop** [comment]
This fixes #153

**Comment by Mossop** [comment]
@ncalexan You might want to look at this too Nick. Now all the main source code is in the `app` directory (maybe we should rename it to `src` at this point). The `README.md` has some notes about what is what there.

**Comment by Mossop** [comment]
So a problem with this is that it then allows us to use node modules inside the renderer UI. It sounds like we don't want that so we might need to stick with webpack.

**Comment by victorporof** [comment]
@Mossop It "allows" us, but we don't have to, right? We can have strict eslint rules that prevent us from accidentally doing that.

**Comment by victorporof** [comment]
Didn't mean to close.

**Comment by victorporof** [comment]
Code looks good, I'd personally like to see this merged.

**Comment by Mossop** [comment]
Yeah I need to get the eslint failures fixed which I think involves writing a new npm module :cry:

**Comment by victorporof** [comment]
I can take over if you don't want to do that 👍 What do you mean by writing a new npm module?

**Comment by Mossop** [comment]
So we transform .jsx -> .js files. To make import statements work in both cases I dropped the ".jsx" part from the import statements. But then the eslint import plugin can't find the .jsx files so it flags a no-unresolved error. Looks like you can extend the plugin with a resolver module to find new files but it looks like no-one has written a resolver for jsx files yet. https://github.com/benmosher/eslint-plugin-import#resolvers

**Comment by Mossop** [comment]
I guess we could just rename the jsx files to js, I don't know if babel cares about the extension. But I bet other tools might

**Comment by victorporof** [comment]
Ugh 😢

**Comment by linclark** [comment]
Not 100% sure, but I've seen `.js` used interchangeably with `.jsx` a lot in the community, so I don't think there are any major tools that depend on the file extension.

**Comment by Mossop** [comment]
Turns out I just needed to read deeper into the docs to find the setting to turn on jsx support!. I'll get this rebased and ready to go

**Comment by Mossop** [comment]
This is updated and should now pass tests

File: app/ui/browser/reducers/main-reducers.js [hunk 1]
```diff
@@ -11,11 +11,12 @@ specific language governing permissions and limitations under the License.
 */
 
 import assert from 'assert';
+import 'babel-polyfill';
```
Comment (victorporof) [inline app/ui/browser/reducers/main-reducers.js]: Why do we need this here?

Comment (Mossop) [inline app/ui/browser/reducers/main-reducers.js]: This file uses Object.entries which isn't supported in node.

File: build/task-build.js [hunk 2]
```diff
@@ -44,6 +27,9 @@ async function buildFile({ path: sourceFile, stats: sourceStats }, targetFile) {
 
   const extension = path.extname(sourceFile);
   if (extension === '.js' || extension === '.jsx') {
+    if (extension === '.jsx') {
+      targetFile = targetFile.substring(0, targetFile.length - 1);
```
Comment (victorporof) [inline build/task-build.js]: This looks a bit terrible. If by any chance we'd like to also support files with the extension .foo at some point, we'd have to substring -3 and add a .js etc.

Extracting the basename and patch and attaching an extension seems like a better way to do this, but I can totally understand why you'd not want to do it now.

File: app/ui/browser/reducers/main-reducers.js (hunk 1 above)
Comment (victorporof) [inline app/ui/browser/reducers/main-reducers.js]: So we'd have to include babel-polyfill in every file that needs a polyfill? That sounds yucky.

**Comment by victorporof** [comment]
Really happy to see this patch. My only real concern is about babel-polyfill.

File: app/ui/browser/reducers/main-reducers.js (hunk 1 above)
Comment (Mossop) [inline app/ui/browser/reducers/main-reducers.js]: Chatted with Jordan and let's just land this for now and we can see if that becomes an issue

Comment (jsantell) [inline app/ui/browser/reducers/main-reducers.js]: Yucky, yeah -- but let's land it and deal with that later. Sounds like we'd either need a build step or not use these new methods in this case

**Comment by victorporof** [comment]
This was merged in https://github.com/mozilla/tofino/commit/3ca2ded897e095aecf321d9d477984fa169855ff

**Pull request closed without merging**